### PR TITLE
fix: Enable fuzzy matching for HRCPrecompileSuite

### PR DESCRIPTION
**Pull request by stoyanov-st**
**Description**:
To match the `mono` behaviour I added a condition in `FinalizeChildRecordHandler` to set `null` to `transferList` when the record failed.

**Related issue(s)**:

Fixes https://github.com/hashgraph/hedera-services/issues/10874

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by tinker-michaelj — CHANGES_REQUESTED** [review]
Quite a resourceful match mode implementation @stoyanov-st!

I think empty transfer lists are likely to appear in production record streams, however; and we should try to make mod-service match mono-service in this case.

Note that transfer lists are constructed and set just in sub-classes of [`RecordFinalizerBase`](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/RecordFinalizerBase.java); so the fix should be pretty targeted.

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/FinalizeChildRecordHandler.java [hunk 1]
```diff
@@ -66,6 +67,8 @@ public void finalizeChildRecord(@NonNull final ChildFinalizeContext context) {
             recordBuilder.transferList(TransferList.newBuilder()
                     .accountAmounts(asAccountAmounts(hbarChanges))
                     .build());
+        } else if (recordBuilder.status() != ResponseCodeEnum.SUCCESS) {
+            recordBuilder.transferList(null);
```
Comment (mhess-swl) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/FinalizeChildRecordHandler.java:71]: `finalizeChildRecord` feels like a weird place to make sure the _user transaction_ transfer list has nothing in it. I'm not sure a better place for it though 🤔

Comment (stoyanov-st) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/FinalizeChildRecordHandler.java:77]: In `mono` when there are multiple failing child records we observe nulled out fields like `transferList` while in `mod` we provide the default values. This difference (mono null vs mod empty objet)  caused the fuzzy-match to fail so I added this change to match the mono behaviour.

Comment (mhess-swl) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/FinalizeChildRecordHandler.java:77]: Nevermind, I was thinking that there was also a `childRecordBuilder` available

**Review by mhess-swl — COMMENTED** [review]
Looks good from services side

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/FinalizeChildRecordHandler.java [hunk 2]
```diff
@@ -72,6 +72,9 @@ public void finalizeChildRecord(@NonNull final ChildFinalizeContext context, fin
             // set an empty transfer list even if there are no hbar changes but the parent record succeeded
             // to be compatible with mono-service
             recordBuilder.transferList(TransferList.DEFAULT);
+        } else if (recordBuilder.status() != ResponseCodeEnum.SUCCESS) {
```
Comment (tinker-michaelj) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/FinalizeChildRecordHandler.java:75]: Is this still required after merging `develop`? C.f. [this change](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleContextImpl.java#L629) which initializes child `transferList` to `null`.

Comment (stoyanov-st) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/FinalizeChildRecordHandler.java:75]: Even with the recent changes the `transferList` is still present in the result.

**Review by tinker-michaelj — APPROVED** [review]
Not sure why this is necessary, but seems reasonable, LGTM @stoyanov-st

**Pull request merged**